src/main/java/camp/nextstep/edu/level1/lotto/lotto/Money.java

```diff
@@ -0,0 +1,73 @@
+package camp.nextstep.edu.level1.lotto.lotto;
+
+import java.util.Objects;
+
+public class Money {
+    private final int MIN_VALUE = 0;
+
+    private final long amount;
+
+    public Money(long amount) {
+        checkValidateMoney(amount);
+        this.amount = amount;
+    }
+
+    public long availablePurchaseCount(long price) {
+        checkPurchasePrice(price);
+        return this.amount / price;
+    }
+
+    public Money add(Money target) {
+        return new Money(this.amount + target.amount);
```

Comment:
```suggestion
        if(target.amount == 0) {
             return this;
        }
        return new Money(this.amount + target.amount);
```

소소하지만 메모리를 절약 할 수 있어요.

Reply:
오..! 감사합니다.

이런 방식이 있군요. ㅎㅎ
다양하게 생각을 하려 하지만.. 습관적으로 코딩을 하곤 했었던 점을 반성하게 되네요